integration-tests/jq/src/main/java/org/apache/camel/quarkus/component/jq/it/JqResource.java

```diff
@@ -0,0 +1,157 @@
+        ObjectNode expected = mapper.createObjectNode().put("foo", "MyValue");
+        endpoint.expectedBodiesReceived(expected);
+
+        ObjectNode node = mapper.createObjectNode().put("foo", "bar");
+        producerTemplate.sendBodyAndHeader("direct:expressionHeaderFunction", node, "MyHeader", "MyValue");
+
+        endpoint.assertIsSatisfied(5000);
+    }
+
+    @Path("/expression/header/string")
+    @GET
+    public void expressionHeaderString() throws Exception {
+        MockEndpoint endpoint = context.getEndpoint("mock:expressionHeaderString", MockEndpoint.class);
+        endpoint.expectedBodiesReceived("bar");
+
+        ObjectNode node = mapper.createObjectNode().put("foo", "bar");
+        producerTemplate.sendBodyAndHeader("direct:expressionHeaderString", null, "Content", node);
+
+        endpoint.assertIsSatisfied(5000);
+    }
+
+    @Path("/expression/pojo")
+    @GET
+    public void expressionPojo() throws Exception {
+        MockEndpoint endpoint = context.getEndpoint("mock:expressionPojo", MockEndpoint.class);
+        endpoint.expectedBodiesReceived(new Book("foo", "bar"));
+
+        ObjectNode node = mapper.createObjectNode();
+        node.with("book").put("author", "foo").put("title", "bar");
+
+        producerTemplate.sendBody("direct:expressionPojo", node);
+
+        endpoint.assertIsSatisfied(5000);
+    }
+
+    @Path("/expression/property")
+    @GET
+    public void expressionProperty() throws Exception {
+        MockEndpoint endpoint = context.getEndpoint("mock:expressionProperty", MockEndpoint.class);
+        endpoint.expectedMessageCount(1);
+
+        ObjectNode node = mapper.createObjectNode().put("foo", "bar");
+
+        producerTemplate.send("direct:expressionProperty", new Processor() {
+            @Override
+            public void process(Exchange exchange) throws Exception {
+                exchange.setProperty("Content", node);
+            }
+        });
+
+        endpoint.assertIsSatisfied(5000);
+    }
+
+    @Path("/expression/property/function")
+    @GET
+    public void expressionPropertyFunction() throws Exception {
+        ObjectNode node = mapper.createObjectNode().put("foo", "MyPropertyValue");
+        MockEndpoint endpoint = context.getEndpoint("mock:expressionPropertyFunction", MockEndpoint.class);
+        endpoint.expectedBodiesReceived(node);
+
+        producerTemplate.send("direct:expressionPropertyFunction", new Processor() {
+            @Override
+            public void process(Exchange exchange) throws Exception {
+                exchange.setProperty("MyProperty", "MyPropertyValue");
+                exchange.getMessage().setBody(mapper.createObjectNode().put("foo", "bar"));
+            }
+        });
+
+        endpoint.assertIsSatisfied(5000);
+    }
+
+    @Path("/filter")
+    @GET
+    public void filter() throws Exception {
+        MockEndpoint endpoint = context.getEndpoint("mock:filter", MockEndpoint.class);
+        endpoint.expectedMessageCount(1);
```

Comment:
What would you think about asserting that the right message was filtered ? I mean like == "valid" ?